services/QuillLMS/app/controllers/api/v1/activity_sessions_controller.rb

```diff
@@ -85,22 +105,15 @@ def destroy
 
   private def activity_session_params
     params.delete(:activity_session)
-    @data ||= params.delete(:data)
-    @time_tracking ||= @data && @data['time_tracking']
-    params.permit(:id,
-                  :access_token, # Required by OAuth
-                  :percentage,
-                  :state,
-                  :question_type,
-                  :completed_at,
-                  :classroom_unit_id,
-                  :activity_uid,
-                  :activity_id,
-                  :anonymous,
-                  :temporary
-                )
-      .merge(data: @data).reject {|k,v| v.nil? }
-      .merge(timespent: @activity_session&.timespent || ActivitySession.calculate_timespent(@time_tracking))
+    data = params.delete(:data)&.permit!
+    time_tracking = data && data['time_tracking']
```

Comment:
Why don't we want the `||=` here?

Reply:
Unless I'm missing something, `||=` isn't necessary since `data` or `time_tracking` shouldn't have values before reaching 108-109.

Reply:
Actually, you're right.  I was only looking at the `update` action but it's called twice in `create`.

Reply:
The solution you have here is fine with me, but I guess it still seems to me like we might as well use `||=` for the sake of idempotency. Is there a reason to avoid it?

Reply:
While the `||=` was definitely needed earlier when the `create` method had a path where `activity_session_params` was called twice, it is only called [once](https://github.com/empirical-org/Empirical-Core/pull/7882/commits/06db1c2a45f16ca87a5b5b2a7f59c6c352e392a2#diff-fb0cb613d7c02517cbea8eab235eb3f3effa4c0340ce0d8929139e8afdb231b7L46) now, and I think that leaving the `||=` would obscure future reading of the code.